xapian-applications/omega/index_file.cc

```diff
@@ -532,8 +533,8 @@ index_add_document(const string & urlterm, time_t last_altered,
 void
 index_mimetype(const string & file, const string & urlterm, const string & url,
 	       const string & ext,
-	       const string &mimetype, DirectoryIterator &d,
-	       Xapian::Document & newdocument,
+	       string &mimetype, DirectoryIterator &d,
```

Comment:
This function can now modify the passed `mimetype` value, but the only caller doesn't require that - it doesn't use `mimetype` at all after calling `index_mimetype()`.  Passing by non-`const` reference tends to lead to surprises so doing it when it isn't necessary is better avoided.

Reply:
Okay, I understand that. I write it as 'non-const' because I need to make changes on them(these changes were previously made on index_file function), but if we use this function in another part of the code i could produce bugs. I will fix it.

Reply:
Indeed.  Or if code is added after the existing call which uses the value of `mimetype`.